packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -354,11 +403,21 @@ export function getInternalReactConstants(
       case HostText:
       case Fragment:
         return null;
+      case FundamentalComponent:
+        return 'Fundamental';
+      case LazyComponent:
+        return 'Lazy';
```

Comment:
Should this not show the inner type? Or does it mean it hasn't yet resolved?

Reply:
I think it means it hasn't resolved yet. Once it resolves, it will be replaced with the inner component:
https://github.com/facebook/react/blob/3f73dcee374732a331b7e72c88ff6dd3b7116a42/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L1119-L1121

So this label likely wouldn't be user-visible, but could show up in the console logs if we enabled the DevTools DEBUG flag for example.

I actually noticed this while tracking down a bug with Lazy (if you unmount a Lazy component before it resolves, React ends up telling DevTools to unmount something that was never mounted and DevTools throws).